import os
import mysql.connector
from mysql.connector import Error
import time

DB_HOST = os.getenv("DB_HOST", "envoy")
DB_PORT = int(os.getenv("DB_PORT", 3307))
DB_USER = os.getenv("DB_USER", "user")
DB_PASSWORD = os.getenv("DB_PASSWORD", "password")
DB_NAME = os.getenv("DB_NAME", "mydatabase")

def connect_to_db():
    while True:
        try:
            connection = mysql.connector.connect(
                host=DB_HOST,
                port=DB_PORT,
                user=DB_USER,
                password=DB_PASSWORD,
                database=DB_NAME
            )

            if connection.is_connected():
                print("Connected to MySQL through Envoy Proxy!")
                cursor = connection.cursor()
                cursor.execute("SELECT * FROM users;")
                for row in cursor.fetchall():
                    print(row)
                cursor.close()
                connection.close()
                break

        except Error as e:
            print(f"Error: {e}. Retrying in 5 seconds...")
            time.sleep(5)

if __name__ == "__main__":
    connect_to_db()

